x/pairing/keeper/pairing.go

```diff
@@ -165,19 +196,26 @@ func (k Keeper) getProjectStrictestPolicy(ctx sdk.Context, project projectstypes
 	}
 
 	if !planstypes.CheckChainIdExistsInPolicies(chainID, policies) {
-		return planstypes.Policy{}, 0, fmt.Errorf("chain ID not found in any of the policies")
+		return planstypes.Policy{}, 0, sdk.Int{}, fmt.Errorf("chain ID not found in any of the policies")
 	}
 
+	spec, found := k.specKeeper.GetSpec(ctx, chainID)
+	if !found {
+		return planstypes.Policy{}, 0, sdk.Int{}, fmt.Errorf("spec not found with chain ID")
+	}
+
+	minStake := spec.MinStakeProvider.Amount
```

Comment:
does not belong to this function, and other places use it, remove it form here to somewhere else, also you are reading the spec already in different places you can save on reading the spec (expensive), specFoundAndActive reads it already

Reply:
not relevant anymore